flake.nix

```diff
@@ -284,143 +186,73 @@
         };
       };
     in
-    rec {
+    {
+      # We use `checks` to run `cargo clippy` and `cargo fmt` since we disable checks in the primary derivations
       checks = {
-        cargo-clippy = craneLib.cargoClippy (nativeArgs // {
-          inherit GIT_COMMIT GIT_DIRTY;
+        cargo-clippy = craneLib.cargoClippy (nativeConfig // {
+          pname = "noir";
 
-          cargoArtifacts = native-cargo-artifacts;
-        });
-
-        cargo-fmt = craneLib.cargoFmt (nativeArgs // {
           inherit GIT_COMMIT GIT_DIRTY;
 
           cargoArtifacts = native-cargo-artifacts;
-          doCheck = true;
         });
 
-        cargo-test = craneLib.cargoTest (nativeArgs // (testArgs 8000) // {
+        cargo-fmt = craneLib.cargoFmt (nativeConfig // {
+          pname = "noir";
+
           inherit GIT_COMMIT GIT_DIRTY;
 
           cargoArtifacts = native-cargo-artifacts;
         });
       };
 
       packages = {
-        default = noir-native;
+        # Nix flakes cannot build more than one derivation in one command (see https://github.com/NixOS/nix/issues/5591)
+        # so we use `symlinkJoin` to build everything as our default package.
+        default = pkgs.symlinkJoin { name = "noir"; paths = [ nargo noir_wasm noirc_abi_wasm ]; };
```

Comment:
cc @kobyhallx for thoughts on the build-all-by-default change

Reply:
let's make nargo as flake default so no extra args are needed to install into profile